### ISSUE TITLE
OnExecute Extension method breaks async actions

### ISSUE THREAD

**Issue by dfbaskin** [natemcmaster/CommandLineUtils#46]
The `OnExecute` extension method provides a way to pass an `Action` delegate rather than a `Func<int>` delegate, but it also breaks if the delegate is an `async` method.

```c#
        /// <summary>
        /// Sets <see cref="CommandLineApplication.Invoke"/> with a return code of <c>0</c>.
        /// </summary>
        /// <param name="app"></param>
        /// <param name="action"></param>
        public static void OnExecute(this CommandLineApplication app, Action action)
            => app.OnExecute(() =>
                {
                    action();
                    return 0;
                });
```

This breaks (because the method is not awaited):

```c#
app.OnExecute(async () => {
    ...
    if(someReason)
        throw new SomeException();
});
```

But this works:

```c#
app.OnExecute(async () => {
    ...
    if(someReason)
        throw new SomeException();
    return 0;
});
```

**Comment by natemcmaster** [comment]
Yup, the compiler picks the wrong overload. Should be pretty easy to workaround in 2.1 by adding your own extension method, like the one added in https://github.com/natemcmaster/CommandLineUtils/commit/8a2f470b7dec9dc9d4b39278e82b23841f0bb4dd. This fix will be in the 2.2 release.

**Comment by dfbaskin** [comment]
Thanks!
